frontend/utils/faviconBadge.js

```diff
@@ -0,0 +1,154 @@
+let currentFavicon = null
```

Comment:
If this code is based on some other project (as it seems to be), we must include attribution, along with a copy of that project's license in this file.

Reply:
@taoeffect  I've used the library 'tinycon.js' as reference, but made many changes. Should I include the license in this case?

Reply:
@Silver-IT Yes, the MIT license requires it: https://github.com/tommoor/tinycon/blob/master/LICENSE

Please also include a link back to the original project. You can also add a comment saying this is a modified version of tinycon.js, but it's important to always credit the original authors.

Reply:
@taoeffect  Can I add the comment below at the top of the file?

/*!
 * FaviconBadge - A small library for manipulating the Favicon
 * Modified from Tinycon.js by Tom Moor
 * Original project: https://github.com/tommoor/tinycon
 * Copyright (c) 2023 Alex Jin
 * @license MIT Licensed
 */

Reply:
@Silver-IT Yep! That would work well too 😄 👍